src/wix/WixToolset.Converters/WixConverter.cs

```diff
@@ -2854,39 +2862,168 @@ private static void RenameElementToStandardDirectory(XElement element)
             }
         }
 
-        private static bool TryGetInnerText(XElement element, out string value, out List<XNode> comments)
-        {
-            return TryGetInnerText(element, out value, out comments, new List<XNode>());
-        }
-
-        private static bool TryGetInnerText(XElement element, out string value, out List<XNode> comments, List<XNode> initialComments)
+        // This function is used to simultaneously extract the inner text of, and any comments embedded in, an XElement.
+        // The comments are returned in a list, so there can be several of them. In addition, the code is designed so that
+        // it can be called in a loop, processing one element at a time, extracting the comments cumulatively, and returning
+        // the inner text separately with each call. An initial list is passed into the "initialComments" parameter. This
+        // parameter defaults to null, which is replaced by an empty list. The augmented list is returned through the "comments"
+        // out parameter. Thiks list is then passed into the "initialComments" paameter the next time around the loop.
+        //
+        // The inner text is returned through the "value" out parameter. The function returns true if inner text is found.
+        //
+        // There are two other parameters that control the behavior of the function: The "protectQuotes" parameter, with a default
+        // value of "true", causes single and double quotes to be processed into an appropriate set of escape sequences (escaped with
+        // a backslash) ap sppropriate for a "Condition" attribute. Contiguous whitespace is coalesced into a single space character
+        // and leading and trailing whitespace is trimmed. When this paarameter is false, the inner text is returned with all whitespace
+        // intact.
+        //
+        // The other optional parameter, "reportWhitespace", has a default value of "false". When it is true, the value returned by the
+        // function is true, not only when non-whitespace text is returned in the "value" out parameter, but also if only whitespace was
+        // found. In this case, the text returned is a single space. If this parameter is false, the text found is trimmed of leading and
+        // trailing whitespace before being returned and the function only returns true if the returned text is not empty.
+        //
+        // There is one other important case in which the function returns true: If a CDATA node is processed, even if all the whitespace removed
+        // causes the remaining text to be empty, the function returns true. If in fact, the remaining text is empty, the text returned is a single
+        // space.
+        private static bool TryGetInnerText(XElement element,
+                                            out string value,
+                                            out List<XNode> comments,
```

Comment:
`out` values should go at the end of the function. Also, we don't wrap parameter lines like this in WiX.

The multiple `out` values suggest that this function is either doing too much and needs to be split, or it should return an object with the data instead of three return parameters.

Reply:
All out parameters removed and replaced by use of a context  parameter.